### ISSUE TITLE
Add support for submodules

### ISSUE THREAD

**Issue by Plaputta** [n0mer/gradle-git-properties#23]
Hi,

could you please add support for submodules? 

I have a main project with several submodules that use gradle-git-properties, trying to build them fails because there is no .git folder inside the submodule:

`09:04:42.158 [ERROR] [org.gradle.BuildExceptionReporter] Execution failed for task ':generateGitProperties'.
09:04:42.160 [ERROR] [org.gradle.BuildExceptionReporter] > org.eclipse.jgit.errors.RepositoryNotFoundException: repository not found: /myProject/myProject-server`

I think changing this `def repo = Grgit.open(dir: project.gitProperties.gitRepositoryRoot ?: project.rootProject.file('.'))` to this `def repo = Grgit.open(currentDir: project.gitProperties.gitRepositoryRoot ?: project.rootProject.file('.'))` may fix the issue, see https://github.com/ajoberstar/grgit/issues/64#issuecomment-106309352

Thank you,
Tobias

**Comment by jesperronn** [comment]
I ran into this same issue for a project with submodules, and I simply cannot build or test any subproject with the `gradle-git-properties` dependency. Very annoying.

Since maintainers did not respond (yet), maybe you can attempt to make a pull request, @Plaputta ?

**Comment by edubkn** [comment]
@Plaputta @jesperronn Were you guys able to sort this out?

**Comment by tha2015** [comment]
It does work with multimodule projects. See attached zip file for an example.
[rep1.zip](https://github.com/n0mer/gradle-git-properties/files/852196/rep1.zip)

I guess you got the error because the root project (top level project, parent of  all sub modules) doesn't contain the .git folder. So you should configure the plugin so it can find .git folder.

In your  case, you should specify the location of the .git folder relatively with the root (top level) project. Below is an example if the .git folder is in a parent folder of the top level project
```
gitProperties {
    gitRepositoryRoot = new File("${project.rootProject.rootDir}/..")
}
```

**Comment by soxsten** [comment]
I have the same problem.
The .git folder of submodule base here - ../../.git/modules/hri-backend/modules/user-service
But when I try: 
**gitRepositoryRoot = new File("${project.rootProject.rootDir}../../.git/modules/hri-backend/modules/user-service")**

I get:
**org.eclipse.jgit.errors.RepositoryNotFoundException: repository not f
ound: C:\IdeaProjects\hri\hri-backend\user-service\.git**

And I can't understand - why it try to find .git folder at _C:\IdeaProjects\hri\hri-backend\user-service\.git_

**Comment by tha2015** [comment]
@soxsten The current version of the plugin doesn't support git modules. I'm submitting  PR #59 to support detect .git files for git modules

**Comment by soxsten** [comment]
@tha2015 
Well... Thank you for the fast respose.
How quickly will you realize this feature?

**Comment by tha2015** [comment]
Feature is implemented in PR #59 and is ready to be reviewed by @n0mer .
We will have to wait for @n0mer  to merge it into the main branch and release it in the next version of the plugin.

**Comment by n0mer** [comment]
@soxsten are you willing to try this feature once it's released?

I'm going to publish new 1.4.22 version soon, once #52 regression is fixed.

Just wondering - do you guys @soxsten @tha2015  use java 8?

**Comment by tha2015** [comment]
I'm using Java 8 (with Gradle) for my everyday job. But I'm also using Java 6 when fixing bugs for an old branch (which is built by Ant, not Gradle).
I guess people using Java 6 not likely build with Gradle. So it might be OK to stop supporting Java version < 8.

**Comment by soxsten** [comment]
@n0mer 
Yes I will. It's very interesting for me. I can wait a bit.
P.S. I use Java 8 with Gradle too.

**Comment by tha2015** [comment]
Code merged. Should be fixed in next version 1.4.22
